src/Components/ServiceScene/Breakdowns/AddToExplorationButton.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useMemo } from 'react';
+import { DataFrame, DataSourceJsonData, TimeRange } from '@grafana/data';
+import { DataSourceWithBackend, usePluginLinks } from '@grafana/runtime';
+import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectState, SceneQueryRunner } from '@grafana/scenes';
+import { DataQuery, DataSourceRef } from '@grafana/schema';
+import { IconButton } from '@grafana/ui';
+import { getLokiDatasource } from 'services/scenes';
+import LokiLogo from '../../../img/logo.svg';
+
+export interface AddToExplorationButtonState extends SceneObjectState {
+  frame?: DataFrame;
+  ds?: DataSourceWithBackend<DataQuery, DataSourceJsonData>;
+  labelName?: string;
+  fieldName?: string;
+
+  disabledLinks: string[];
+}
+
+type ExtensionContext = {
+  timeRange: TimeRange;
+  queries: DataQuery[];
+  datasource: DataSourceRef;
+  origin: string;
+  url: string;
+  type: string;
+  title: string;
+  id: string;
+  logoPath: string;
+  note?: string;
+  drillDownLabel?: string;
+};
+
+export class AddToExplorationButton extends SceneObjectBase<AddToExplorationButtonState> {
+  constructor(state: Omit<AddToExplorationButtonState, 'disabledLinks'>) {
+    super({ ...state, disabledLinks: [] });
+    this.addActivationHandler(this.onActivate);
+  }
+
+  private onActivate = () => {
+    getLokiDatasource(this).then((ds) => {
+      this.setState({ ds });
+    });
+  };
+
+  public static Component = ({ model }: SceneComponentProps<AddToExplorationButton>) => {
+    const { ds, frame, labelName, fieldName, disabledLinks } = model.useState();
+
+    const data = sceneGraph.getData(model);
+    const sqr = sceneGraph.findObject(data, (o) => o instanceof SceneQueryRunner) as SceneQueryRunner;
+
+    const queries = useMemo(() => {
+      return sqr?.state.queries.map((q) => ({
+        ...q,
+        expr: sceneGraph.interpolate(sqr, q.expr),
+        legendFormat: sceneGraph.interpolate(sqr, q.legendFormat),
+      }));
+    }, [sqr]);
+
+    const datasourceUid = ds?.uid;
+    const timeRange = sceneGraph.getTimeRange(model);
+
+    useMemo(() => {
+      if (frame) {
+        const filter = getFilter(frame);
+        queries?.forEach((query: DataQuery & { legendFormat: string }) => {
+          if (filter) {
+            query.legendFormat = `{{${filter.name}}}`;
+          }
+        });
+      }
+    }, [frame, queries]);
+
+    const extensionPointId = 'grafana-lokiexplore-app/metric-exploration/v1';
```

Comment:
I know this is a WIP, but this is a lot of work happening in the render body of the component. If you can extract some of these to functions and constants (like this extensionPointId) it would be great.